scripts/o3de/o3de/engine_template.py

```diff
@@ -1240,6 +1240,15 @@ def create_from_template(destination_path: pathlib.Path,
         logger.error(f'Template Name or Template Path must be specified.')
         return 1
 
+    # if replacements are provided we need an even number of
+    # replace arguments because they are A->B pairs
+    if replace and len(replace) % 2 == 1:
+        logger.error("Invalid replacement argument pairs.  "
+                     "Verify you have provided replacement match and value pairs "
+                     "and your replacement arguments are in single quotes "
+                     "e.g. -r '${GemName}' 'NameValue'")
```

Comment:
Can you also log the replacement pairs as well?
```suggestion
    if replace and len(replace) % 2 == 1:
        replacement_pairs = []
        for replace_index in range(0, len(replace), 2):
            if replace_index + 1 < len(replace):
                replacement_pairs.append(f'{replace[replace_index]} -> {replace[replace_index + 1]}')
            else:
                replacement_pairs.append(f'{replace[replace_index]} is missing replacement')
        logger.error("Invalid replacement argument pairs.  "
                     "Verify you have provided replacement match and value pairs "
                     "and your replacement arguments are in single quotes "
                     "e.g. -r '${GemName}' 'NameValue'\n"
                     "The current set of replacement pairs are as follows:\n"
                     "\n".join(replacement_pairs))
```

Reply:
Good idea